contracts/lib/MerklePatriciaProof.sol

```diff
@@ -60,6 +60,7 @@ library MerklePatriciaProof {
                 nodeKey = RLP.toBytes32(currentNodeList[nextPathNibble]);
                 pathPtr += 1;
             } else if(currentNodeList.length == 2) {
+                uint currentpathPtr = pathPtr;
```

Comment:
Wouldn't it be more intuitive to keep the `pathPtr` and instead introduce a new variable for the leaf node check? (e.g. `advancedPathPointer` or something with a better name.

Reply:
Would that not be confusing ? 🤔 
It would look like this. In below snippet, again `pathPtr` needs to be updated with `advancedPathPointer`

```
                uint advancedPathPointer = pathPtr;
                advancedPathPointer += _nibblesToTraverse(RLP.toData(currentNodeList[0]), path, pathPtr);

                if(pathPtr == path.length) {//leaf node
                    if(keccak256(abi.encodePacked(RLP.toData(currentNodeList[1]))) == value) {
                        return true;
                    } else {
                        return false;
                    }
                }
                //extension node ... test if means that it is empty value
                if(_nibblesToTraverse(RLP.toData(currentNodeList[0]), path, pathPtr) == 0) {
                    return false;
                }
                pathPtr = advancedPathPointer;
```

Reply:
"advanced path pointer" would be required in the leaf node check and the 0 length check. Anyways, I prefer Ben's suggestion, actually.